test/grizzly/zwave_firmware_test.exs

```diff
@@ -0,0 +1,267 @@
+defmodule Grizzly.ZWaveFirmwareTest do
+  use ExUnit.Case, async: false
+  use Mimic
+
+  setup :set_mimic_global
+
+  alias Grizzly.{Options, ZWaveFirmware, ZWaveFirmware.UpgradeSpec}
+
+  @corrupt_fw_output """
+  Using serial device /dev/tty.usbmodem0004402548831
+  SerialAPI: Retransmission 0 of 0x07
+  SerialAPI: Retransmission 1 of 0x07
+  SerialAPI: Retransmission 2 of 0x07
+  SerialAPI: Retransmission 3 of 0x07
+  SerialAPI: Retransmission 4 of 0x07
+  SerialAPI: Retransmission 5 of 0x07
+  SerialAPI: Retransmission 6 of 0x07
+  SerialAPI: Retransmission 7 of 0x07
+  SerialAPI: Reopening serial port
+  """
+
+  describe "maybe_run_zwave_firmware_update/1" do
+    test "when disabled" do
+      stub(MuonTrap)
+      stub(MockStatusReporter)
+      stub(MockZWaveResetter)
+
+      ZWaveFirmware.maybe_run_zwave_firmware_update(get_options(%{enabled: false}))
+    end
+
+    test "normal operation" do
+      pid = self()
+
+      MuonTrap
+      |> expect(:cmd, fn "/usr/bin/zw_programmer", args ->
+        send(pid, args)
+        {zw_programmer_version_output("7.18.02"), 0}
+      end)
+      |> expect(:cmd, fn "/usr/bin/zw_programmer", args ->
+        send(pid, args)
+        {"", 0}
+      end)
```

Comment:
```suggestion
      |> expect(:cmd, fn "/usr/bin/zw_programmer", ["-s", "/dev/null", "-t"] ->
        {zw_programmer_version_output("7.18.02"), 0}
      end)
      |> expect(:cmd, fn "/usr/bin/zw_programmer", ["-s", "/dev/null", "-p", "/path/to/7.19.3.gbl"] ->
        {"", 0}
      end)
```
Maybe you have reasoning for this, but you can also include the args in your expectation rather than rely on sending a message back to the test process. But FYI in case that might be preferred

Reply:
I could've sworn that wasn't working on Friday, but I must've been doing something wrong. 🙃